pallets/xyk/src/lib.rs

```diff
@@ -2303,18 +2314,36 @@ impl<T: Config> XykFunctionsTrait<T::AccountId> for Pallet<T> {
 		Pallet::<T>::settle_treasury_and_burn(sold_asset_id, buy_and_burn_amount, treasury_amount)?;
 
 		if bought_asset_amount < min_amount_out {
-			return Err(DispatchError::from(Error::<T>::InsufficientOutputAmount))
+			if err_upon_bad_slippage {
+				return Err(DispatchError::from(Error::<T>::InsufficientOutputAmount))
+			} else {
+				Pallet::<T>::deposit_event(Event::SellAssetFailed(
+					sender,
+					sold_asset_id,
+					sold_asset_amount,
+					bought_asset_id,
+					bought_asset_amount,
+					min_amount_out,
+				));
+			}
 		}
 
 		Ok(())
 	}
 
+	// To put it comprehensively the only reason that the user should lose out on swap fee
```

Comment:
Not sure if achievable, but would it be possible to align this comment with autogenerated RustDoc format we are currently using? cc: @mateuszaaa

Reply:
Ok, @mateuszaaa explained that it would be there like this. So ignore my comment.